packages/fast-components/src/slider/slider.ts

```diff
@@ -0,0 +1,327 @@
+import { attr, FastElement, observable, Observable } from "@microsoft/fast-element";
+import { FormAssociated } from "../form-associated";
+import {
+    keyCodeArrowRight,
+    keyCodeArrowLeft,
+    keyCodeArrowDown,
+    keyCodeArrowUp,
+} from "@microsoft/fast-web-utilities";
+import { convertPixelToPercent } from "./slider-utilities";
+
+export enum SliderMode {
+    singleValue = "single-value",
+    adjustFromLower = "adjust-from-lower",
+    adjustFromUpper = "adjust-from-upper",
+    dualThumb = "dual-thumb",
+}
+
+export enum SliderOrientation {
+    horizontal = "horizontal",
+    vertical = "vertical",
+}
+
+export class Slider extends FormAssociated<HTMLInputElement> {
+    @attr({ attribute: "readonly", mode: "boolean" })
+    public readOnly: boolean; // Map to proxy element
+    private readOnlyChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.readOnly = this.readOnly;
+        }
+
+        this.readOnly
+            ? this.classList.add("readonly")
+            : this.classList.remove("readonly");
+        this.setAttribute("aria-readonly", this.readOnly.toString());
+    }
+
+    @observable
+    public root: HTMLDivElement;
+
+    @observable
+    public track: HTMLDivElement;
+
+    @observable
+    public foregroundTrack: HTMLDivElement;
+
+    @observable
+    public thumb: HTMLDivElement;
+
+    @observable
+    public upperThumb: HTMLDivElement;
+
+    @observable
+    public lowerThumb: HTMLDivElement;
+
+    @observable
+    public direction: string = "ltr";
```

Comment:
Should this be a string or map to a global enum? I believe we have direction in `fast-web-utilities`

Reply:
Per above, you can remove `@observable` from anything that is a `ref`.

Reply:
done